app/jobs/imo/invite_unsubscribed_patients.rb

```diff
@@ -0,0 +1,27 @@
+class Imo::InviteUnsubscribedPatients
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :default
+
+  REINVITATION_BUFFER = 6.months.freeze
+
+  def perform
+    return unless Flipper.enabled?(:imo_messaging)
+
+    next_messaging_time = Communication.next_messaging_time
+    patients.each do |patient|
+      Imo::InvitePatient.perform_at(next_messaging_time, patient.id)
+    end
+  end
+
+  def patients
+    Patient
+      .contactable
+      .not_ltfu_as_of(Time.current)
+      .left_joins(:imo_authorization)
+      .where(
+        "imo_authorizations.last_invited_at < ? AND imo_authorizations.status != ? OR imo_authorizations.id IS NULL",
```

Comment:
Can you add some parentheses around the AND clause here to make this more explicit?

Reply:
Yes, I had them at one point and must have removed them while testing the query. Totally agree that this needs them.